src/utils/blocksToWeeks.ts

```
@@ -0,0 +1,38 @@
+import { add, differenceInWeeks } from "date-fns";
+
+const BLOCKS_PER_HOUR = 6;
+
+/**
+ * Converts a number of blocks into weeks with optional rounding precision.
+ *
+ * @param {number} blocks - The number of blocks to convert.
+ * @param {number} [precision] - The precision to round the result to. If not provided, the result is not rounded.
+ * @returns {string} - The converted time in weeks, rounded to the specified precision.
+ *
+ * Examples of usage:
+ * blocksToWeeks(30000); // "29 weeks"
+ * blocksToWeeks(63000, 5); // "65 weeks"
+ */
+export const blocksToWeeks = (blocks: number, precision?: number): string => {
+  // If no blocks are provided, return "-"
+  if (!blocks) return "-";
```

Let's throw an error here, it should not be called with an empty parameter.